### PR TITLE
[wip] - CS 5844 (dont merge!)

### PR DESCRIPTION
**Pull request by m038**

### CODE REVIEW

File: newscoop/template_engine/classes/CampURI.php [hunk 1]
```diff
@@ -1020,6 +1020,26 @@ public function isSSL()
...
+     *                                       in the resulsts.
+     *
+     * @return boolean
+     */
+    public function isHomepage($withLanguageHomepage = false)
+    {
+        if ($this->path === '/') {
+            return true;
+        } elseif ($withLanguageHomepage && preg_match('@^/[a-z]{2}(?:_[A-Z]{2})?$@', $this->path)) {
+            // TODO: Check if we need to validate language here
```
Comment (ahilles107) [inline newscoop/template_engine/classes/CampURI.php]: we need

File: newscoop/src/Newscoop/NewscoopBundle/Form/Type/PublicationOutputSettingsType.php [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+namespace Newscoop\NewscoopBundle\Form\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolverInterface;
+use Doctrine\ORM\EntityRepository;
+use Symfony\Component\Validator\Constraints\Length;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+class PublicationOutputSettingsType extends AbstractType
```
Comment (ahilles107) [inline newscoop/src/Newscoop/NewscoopBundle/Form/Type/PublicationOutputSettingsType.php]: @m038 - why we need it? Looks like duplicate of existing PublicationSettingsType

**Comment by ahilles107** [comment]
And i think that this should be done agains master - not 4.4 as it's not a bugfix or even minor feature. I'm right? @gideonlehmann ?

File: newscoop/src/Newscoop/NewscoopBundle/Form/Type/PublicationOutputSettingsType.php (hunk 2 above)
Comment (m038) [inline newscoop/src/Newscoop/NewscoopBundle/Form/Type/PublicationOutputSettingsType.php]: i think this is the same file... i dont have it anymore in my branch...

**Comment by m038** [comment]
Well, you can update it for master! :)

**Comment by gideonlehmann** [comment]
Hey Pawel,

yes, we said that is too big of a change to be regarded as bugfix. But it
would be great to have it available soon, since the Corners people and
afghanistan today want to use storyteller in parallel to their regular
instances.

All the best Gideon

On Wed, Oct 28, 2015 at 12:00 PM, Paweł Mikołajczuk <
notifications@github.com> wrote:

> And i think that this should be done agains master - not 4.4 as it's not a
> bugfix or even minor feature. I'm right? @gideonlehmann
> <https://github.com/gideonlehmann> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sourcefabric/Newscoop/pull/1273#issuecomment-151801276>
> .
>



-- 

Gideon Lehmann

Head of Client Projects, Sourcefabric
gideon.lehmann@sourcefabric.org
www.sourcefabric.org



Prinzessinnenstrasse 20, 10969 Berlin, Germany

+49 (0) 30 616 29 281
+49 (0) 177 3121010
skype: gideon.lehmann

File: newscoop/library/Newscoop/Entity/Repository/IssueRepository.php [hunk 3]
```diff
@@ -114,12 +114,16 @@ public function getIssue($languageCode, $publication, $shortName = null)
...
-            ->select('i.id')
+            ->select('i')
             ->where('i.publication = :publicationId')
+            ->andWhere('i.workflowStatus = :publishStatus')
+            ->andWhere('i.language = :languageId')
             ->setParameter('publicationId', $publicationId)
+            ->setParameter('publishStatus', \Newscoop\Entity\Issue::STATUS_PUBLISHED)
+            ->setParameter('languageId', $languageId)
             ->setMaxResults(1)
             ->orderBy('i.id', 'DESC');
```
Comment (m038) [inline newscoop/library/Newscoop/Entity/Repository/IssueRepository.php]: We should change id to publication date.

**Comment by ahilles107** [comment]
closed by #1295

**Pull request closed without merging**